Source/WebCore/inspector/InspectorFrontendClientLocal.cpp

```diff
@@ -290,8 +290,9 @@ void InspectorFrontendClientLocal::openURLExternally(const String& url)
     if (!frame)
         return;
 
-    frame->setOpener(mainFrame.ptr());
+    ASSERT(frame->opener() == mainFrame.ptr());
     frame->page()->setOpenedByDOM();
+    frame->page()->setOpenedByDOMWithOpener(true);
```

Comment:
This feels like an odd API, in that, `setOpenedByDOMWithOpener` sounds like a superset of `setOpenedByDOM` but `setOpenedByDOMWithOpener` doesn't set `m_openedByDOM`. Are `m_openedByDOM` and `m_openedByDOMWithOpener` really a tri-state?

Reply:
It is odd, but they are distinct, and also distinct from just having an opener.  I think I'm on my way to removing setOpenedByDOMWithOpener, but that can't be done in this PR.